src/AclExtras.php

```diff
@@ -294,7 +294,11 @@ protected function _checkMethods($className, $controllerName, $node, $pluginPath
         $excludes = $this->_getCallbacks($className, $pluginPath);
         $baseMethods = get_class_methods(new Controller);
         $namespace = $this->get_namesapce($className, $pluginPath);
-        $actions = get_class_methods(new $namespace);
+        $reflectionClass = new \ReflectionClass( $namespace );
+        $actions = null;
+        if ($reflectionClass->IsInstantiable()) {
+            $actions = get_class_methods(new $namespace);
```

Comment:
It would fail only if constructor throws something at you, this is probably what you meant by required parameters. I think that missing parameters will only generate warnings.

You are right that ReflectionClass is way to go, and we already have it in this soup.

I'll fix it.